### PR TITLE
Code clean-up

### PR DESCRIPTION
**Pull request by ldecicco-USGS**
Starting down the path to write a JS script...

### CODE REVIEW

File: layout/css/svg.css [hunk 1]
```diff
@@ -3,24 +3,29 @@
...
-  fill:none;
-  stroke:#e0e0e0;
-  stroke-width:0.5;  
+.axis-labels {
+  font-size: 8px;
+  font-family: Arial;
 }
 
 .years-rect {
-  fill:"red";
```
Comment (jordansread) [inline layout/css/svg.css:44]: my bad

File: scripts/process/process-hydrograph.R [hunk 2]
```diff
@@ -1,82 +1,79 @@
...
+    rects <- xml_find_all(rect_svg_all, '//*[local-name()="rect"]')
+    rect_svg <- rects[length(rects)]
     
-    xml_set_attr(rect_svg, 'id', yr)
+    xml_set_attr(rect_svg, 'id', paste0("b",yr))
     xml_attr(rect_svg, "style") <- NULL
     xml_attr(rect_svg, "clip-path") <- NULL
     xml_attr(rect_svg, "class") <- "years-rect"
+    xml_attr(rect_svg, "onmouseover") <- paste0("vizlab.showline('",yr,"')")
+    xml_attr(rect_svg, "onmouseout") <- paste0("vizlab.greyline('",yr,"')")
```
Comment (jordansread) [inline scripts/process/process-hydrograph.R]: do we need a `vizlab.greyline()` function?

File: layout/js/gages.js [hunk 3]
```diff
@@ -110,4 +110,26 @@ var hovertext;
       mapSVG.hideTooltip();
     }
   }
+  
+  vizlab.showline = function(year) {
```
Comment (jordansread) [inline layout/js/gages.js]: I think most of the code in this function is from the fanciness that needed to happen to filter which points are rendered and rebuild the path accordingly to truncate the path to only display sites in a given year. 

I don't know the fancy jquery way to do this, but here is a one way:
```js
vizlab.showline = function(year) {
   var lineId = '#l' + year;
   document.getElementById(lineId).setAttribute(class, 'selected-year');
}
vizlab.greyline = function(year) {
   var lineId = '#l' + year;
   document.getElementById(lineId).setAttribute(class, 'not-sure-what-class');
}
```

File: scripts/process/process-hydrograph.R [hunk 4]
```diff
@@ -1,82 +1,79 @@
...
+      rect(xleft=yr,
+         xright=yr+1,
+         ybottom=0, ytop=yMax)
+    },  height=height, width=width)
+  
+    rects <- xml_find_all(rect_svg_all, '//*[local-name()="rect"]')
+    rect_svg <- rects[length(rects)]
     
-    xml_set_attr(rect_svg, 'id', yr)
+    xml_set_attr(rect_svg, 'id', paste0("b",yr))
```
Comment (jordansread) [inline scripts/process/process-hydrograph.R:64]: Just found out that `xml_set_attrs()` lets you set attrs and drop all of the ones you don't set at the same time:
```r
xml_set_attrs(rect_svg, c("id" = paste0("b",yr), "onmouseover" = paste0("vizlab.showline('",yr,"')"),"onmouseout" =paste0("vizlab.greyline('",yr,"')"))
```
can take the place of lines 58-63

**Review by jordansread — APPROVED** [review]
had a js function suggestion otherwise looks good!

File: scripts/process/process-hydrograph.R (hunk 4 above)
Comment (ldecicco-USGS) [reply scripts/process/process-hydrograph.R:64]: schnazzy

File: scripts/process/process-hydrograph.R (hunk 2 above)
Comment (ldecicco-USGS) [reply scripts/process/process-hydrograph.R]: I don't know, was just starting to play around and didn't realize I committed...whoops

**Comment by jordansread** [comment]
Ready for merge?

File: scripts/process/process-hydrograph.R (hunk 4 above)
Comment (ldecicco-USGS) [reply scripts/process/process-hydrograph.R:64]: Ack, so just fyi, that also takes out the height width, etc.

**Comment by ldecicco-USGS** [comment]
I think it's OK to merge. I'm planning to do some work to get the axis lines/ticks back in a group. The JS function caused all JS to break. So that still needs work. But, I think the stuff that's currently committed is better than what it replaces.

File: layout/js/gages.js [hunk 5]
```diff
@@ -107,4 +107,11 @@ var hovertext;
       mapSVG.hideTooltip();
     }
   }
+  
+//  vizlab.showline = function(year) {
+//     var lineId = '#l' + year;
+//     document.getElementById(lineId).setAttribute(class, 'selected-year'//);
```
Comment (jordansread) [inline layout/js/gages.js:113]: @ldecicco-USGS `selected-year` is the same class used in the other svg, and I think it gets wiped every timestep that the animation plays. Is this the issue? If a different class was created an used, would this still need to be removed?

File: scripts/process/process-hydrograph.R (hunk 4 above)
Comment (jordansread) [reply scripts/process/process-hydrograph.R:64]: That's a gotcha! whoops.

File: layout/js/gages.js (hunk 5 above)
Comment (mbucknell) [reply layout/js/gages.js:113]: One advantage of using jquery and addClass/removeClass is that the function only adds/removes the class string specified. Other classes are left in place. So on the other svg, all of the gage rectangles have class of .gage-count-bar. I add the class "selected-year" to give the red styling. And then remove it when the bar is no longer hovered over. To do the same with straight js, you would need to retrieve the class and add the new one separating the two classes with a space.

**Pull request merged**